src/App/component/PullRequestsPage.jsx

```diff
@@ -0,0 +1,170 @@
+import { useEffect, useState } from 'react'
+import { makeStyles } from '@material-ui/core/styles'
+import ProjectAvatar from './ProjectAvatar'
+import DrawingBoard from './DrawingBoard'
+import Axios from 'axios'
+import moment from 'moment'
+import { Backdrop, CircularProgress } from '@material-ui/core'
+import { connect } from 'react-redux';
+
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    display: 'flex',
+    '& > *': {
+      margin: theme.spacing(1),
+    },
+    minWidth: '30px',
+  },
+  backdrop: {
+    zIndex: theme.zIndex.drawer + 1,
+    color: '#fff',
+  },
+}))
+
+function PullRequestsPage(prop) {
+  const classes = useStyles()
+  const [pullRequestListData, setPullRequestListData] = useState([])
+  const [dataForPullRequestChart, setDataForPullRequestChart] = useState({ labels: [], data: { opened: [], closed: [], merged: [] } })
+
+  const [currentProject, setCurrentProject] = useState({})
+
+  const projectId = localStorage.getItem("projectId")
+  const jwtToken = localStorage.getItem("jwtToken")
+
+  const [open, setOpen] = useState(false);
+  const handleClose = () => {
+    setOpen(false);
+  };
+  const handleToggle = () => {
+    setOpen(!open);
+  };
+
+  useEffect(() => {
+    Axios.get(`http://localhost:9100/pvs-api/project/1/${projectId}`,
+      { headers: { "Authorization": `${jwtToken}` } })
+      .then((response) => {
+        setCurrentProject(response.data)
+      })
+      .catch((e) => {
+        alert(e.response?.status)
+        console.error(e)
+      })

```

Comment:
This code is really impressive.
Because it may be the last few reviews, although this is just for school homework, not a formal open source project, I still want to provide some suggestions, hoping to do better in the future.

- Suggest extracting these API calling processes.
  For a long time, this process should be packaged independently, and in theory, the entire PVS should only be called once, rather than "call just we want", which is actually a waste of time and efficiency.
  But since we are taking things written by others and continuing to do it, I think we don’t need to refactor the whole project because of time.
  But at least the newly added pages are something we can take into account, so try to do a little better.
  In summary, the process of these API calls can be encapsulated.
  You may see there are similar structures in many places...

- Suggest giving a name for every process function. If some code does not have a function yet, suggest creating one.
  Fragmented code makes it difficult to understand the meaning and purpose of execution.